### DISCUSSION TITLE
Unable to export traces from angular web app to a jaeger backend/collector.

### DISCUSSION THREAD

**Discussion by ksg14** [jaegertracing/jaeger#3483] [General]
Unable to export traces from angular web app to a jaeger backend/collector.
1. HTTP Err: 400 for a zipkin exporter
2. HTTP Err: 501 for an OTLP exporter

Steps to reproduce 400 error:
1. Used Zipkin Exporter from @opentelemetry/exporter-zipkin on angular side.
2. Using the following docker run command:
`docker run -d --name jaeger \
  -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
  -p 5775:5775/udp \
  -p 6831:6831/udp \
  -p 6832:6832/udp \
  -p 5778:5778 \
  -p 16686:16686 \
  -p 14250:14250 \
  -p 14268:14268 \
  -p 14269:14269 \
  -p 9411:9411 \
  jaegertracing/all-in-one:1.29`

Steps to reproduce 501 error:
1. OTLP exporter from @opentelemetry/exporter-trace-otlp-http on angular side.
2. Using the following docker run command:
`docker run -d --name jaeger \
  -p 5775:5775/udp \
  -p 6831:6831/udp \
  -p 6832:6832/udp \
  -p 5778:5778 \
  -p 16686:16686 \
  -p 14268:14268 \
  -p 14250:14250 \
  -p 9411:9411 \
  -p 55680:55680 \
  -p 55681:55681 \
  -p 4318:55680 \
  jaegertracing/opentelemetry-all-in-one`

**Expected behavior**
Traces to be exported to collector.

**Screenshots**
![9B789BD1-3852-414E-90FD-B58589DAE4DD_1_105_c](https://user-images.githubusercontent.com/18640573/149168692-023a699f-1c59-420e-8b2d-7d3d58d73394.jpeg)

![8D2EBF57-A094-431F-9E45-22A730EA79C4_1_105_c](https://user-images.githubusercontent.com/18640573/149168723-cc65d5aa-3a5f-4035-8535-029b0fff950e.jpeg)

 - OS: MacOS
 - Jaeger version: 1.29
 - Deployment: Docker

**Comment by pavolloffay** [comment]
> HTTP Err: 501 for an OTLP exporter

Jaeger does not support OTLP protocol at the moment. If you want to use OTLP send data first to OTEL collector and then from collector to Jaeger.


The response from Zipkin exporter shows 400 Bad Request. it is most likely related to cors configuration. Please take a look at

```
      --collector.zipkin.allowed-headers string                   Comma separated list of allowed headers for the Zipkin collector service, default content-type (default "content-type")
      --collector.zipkin.allowed-origins string                   Comma separated list of allowed origins for the Zipkin collector service, default accepts all (default "*")
      --collector.zipkin.host-port string                         The host:port (e.g. 127.0.0.1:9411 or :9411) of the collector's Zipkin server (disabled by default)

```

**Comment by pavolloffay** [comment]
I am closing this as it does not seem like a Jaeger bug but rather configuration issue.

**Comment by ksg14** [comment]
I tried the flags and am still getting the 400 err. Also, Is there another way to export jaeger tracer from an angular app?

**Comment by yurishkuro** [comment]
I've seen a few deployments where people export telemetry from client apps (mobile or web), and in all those cases the architecture was to have a dedicated endpoint implemented by __your__ backend to receive telemetry and then forward that telemetry to the monitoring backends. This way you can reuse all the infra you may already have in place between the client app and your backend, such as auth, regional LBs, DoS prevention, whatever else. The telemetry endpoint acts as just a proxy.

**Comment by dhanaraj-ally** [comment]
@pavolloffay  
Is there any change since your comment? Does Jaeger support HTTP OTLP protocol natively now? I see some blog posts pop up on the same. Port 55680 is being used for sending trace data using OTLP protocol to jaegertracing/all-in-one.
https://rehansaeed.com/exporting-open-telemetry-data-to-jaeger/
